R/sampler-random-walk.R

```diff
@@ -7,14 +7,39 @@
 ##'
 ##' @param vcv A variance covariance matrix for the proposal.
 ##'
+##' @param boundaries Control the behaviour of proposals that are
+##'   outside the model domain.  The supported options are:
+##'
+##'   * "reflect" (the default): we reflect proposed parameters that
+##'     lie outside the domain back into the domain (as many times as
+##'     needed)
+##'
+##'   * "reject": we do not evaluate the density function, and return
+##'     `-Inf` for its density instead.
+##'
+##'   * "ignore": evaluate the point anyway, even if it lies outside
+##'     the domain.
+##'
+##' The initial point selected will lie within the domain, as this is
+##' enforced by [mcstate_sample].
+##'
 ##' @return A `mcstate_sampler` object, which can be used with
 ##'   [mcstate_sample]
 ##'
 ##' @export
-mcstate_sampler_random_walk <- function(vcv = NULL) {
+mcstate_sampler_random_walk <- function(vcv = NULL, boundaries = "reflect") {
   check_vcv(vcv, allow_3d = TRUE, call = environment())
   internal <- new.env()
 
+  ## We'll also support rejection here but this requires a little work
+  ## in the interface for the multiple-parameter set runner, so that
+  ## we can say "only run this subset please".  We need this for the
+  ## PT algorithms too, in order to not evaluate likelihood for the
+  ## sampling directly from the prior, and this is its own piece of
+  ## work really, which I might start in dust, and then we'll formalise
+  ## as part of the expected model interface.
```

Comment:
Do you need this comment now reject has been included as an option?

Reply:
oops, thanks